packages/sqlite-sync/src/index.ts

```diff
@@ -0,0 +1,145 @@
+import {escapeSQLiteIdentifier} from '@databases/escape-identifier';
+import sql, {SQLQuery, isSqlQuery, FormatConfig} from '@databases/sql';
+import DatabaseConstructor, {Statement, Options as DatabaseOptions, Database} from 'better-sqlite3';
+
+export type {SQLQuery};
+export {sql, isSqlQuery};
+
+const sqliteFormat: FormatConfig = {
+  escapeIdentifier: (str) => escapeSQLiteIdentifier(str),
+  formatValue: (value) => ({placeholder: '?', value}),
+};
+
+export interface DatabaseTransaction {
+  query(query: SQLQuery): any[];
+  queryStream(query: SQLQuery): IterableIterator<any>;
+}
+export interface DatabaseConnection extends DatabaseTransaction {
+  tx<T>(fn: (db: DatabaseTransaction) => Promise<T>): Promise<T>;
+  dispose(): void;
+}
+
+class DatabaseTransactionImplementation implements DatabaseTransaction {
+  private readonly _database: Database;
+  constructor(database: Database) {
+    this._database = database;
+  }
+
+  query(query: SQLQuery) {
+    if (!isSqlQuery(query)) {
+      throw new Error('Expected query to be an SQLQuery');
+    }
+    return runQuery(query, this._database);
+  }
+
+  /**
+   * @deprecated use queryStream
+   */
+  queryStream(query: SQLQuery): IterableIterator<any> {
+    if (!isSqlQuery(query)) {
+      throw new Error('Expected query to be an SQLQuery');
+    }
+    return runQueryStream(query, this._database);
+  }
+}
+
+export const IN_MEMORY = ':memory:';
+class DatabaseConnectionImplementation implements DatabaseConnection {
+  private readonly _database: Database;
+  private readonly _begin : Statement;
+  private readonly _commit : Statement;
+  private readonly _rollback : Statement;
+  private lockedPromise : Promise<any> | null;
+  constructor(filename: string, options: DatabaseOptions = {}) {
+    this._database = new DatabaseConstructor(filename, options);
+    this._begin = this._database.prepare('BEGIN');
+    this._commit = this._database.prepare('COMMIT');
+    this._rollback = this._database.prepare('ROLLBACK');
+    this.lockedPromise = null;
+  }
+  query(query: SQLQuery) {
+    if (!isSqlQuery(query)) {
+      throw new Error('Expected query to be an SQLQuery');
+    }
+    return runQuery(query, this._database);
```

Comment:
Please add a test case that runs multiple statements in one go. i.e. it should call:

```ts
db.query(sql`STATEMENT 1; STATEMENT 2`)
```

Where `STATEMENT 1` and `STATEMENT 2` are replaced with valid SQL statements. If this isn't handled out of the box (I expect it won't be) we will need to use `@databases/split-sql-query` to detect this and split the queries up.